libr/asm/asm.c

```diff
@@ -1098,15 +1098,15 @@ R_API RAsmCode* r_asm_rasm_assemble(RAsm *a, const char *buf, bool use_spp) {
 	if (use_spp) {
 		Output out;
 		out.fout = NULL;
-		out.cout = r_strbuf_new ("");
-		r_strbuf_init (out.cout);
+		out.cout = (SStrBuf *)r_strbuf_new ("");
```

Comment:
the right fix is to make spp use r_strbuf from r2 instead of its embedded implementation, otherwise we duplicate functionality or we use different apis with the wrong cast, i reintroduced this "issue" because spp and r2 repos wasnt synced properly and nothing was building, so i adjusted the code to make it work again, but left this warning to be fixed. and in fact as long as the use of s_strbuf is internal to this logic (outside there's just an strdup)

Reply:
I thought in fact it was wanted to use the definition from spp nevertheless, ok will take in account.

Reply:
this just asks for troubles imho...